src/coreclr/inc/cordebuginfo.h

```diff
@@ -431,4 +431,30 @@ class ICorDebugInfo
         // Source information about the IL instruction in the inlinee
         SourceTypes Source;
     };
+
+    struct AsyncContinuationVarInfo
+    {
+        // IL number of variable (or one of the special IL numbers, like UNKNOWN_ILNUM)
+        uint32_t VarNumber;
+        // Offset in continuation's data where this variable is stored
+        uint32_t Offset;
```

Comment:
We probably need to include `GCData` offset here too, but I expect that we will end up with customized continuation layouts during .NET 11 that gets rid of the `Data`/`GCData` split, so I designed some of these types around that expectation. It also should be possible to derive the GCData offset by iterating the vars linearly, so we can probably make do with that for now.

Reply:
> Just in case we may end up using more than one shape, perhaps add some versioning/flavor marker in the AsyncSuspensionPoint?

I expect we stabilize on something during .NET 11, so if we change it after I think we can take an R2R version update at the same time as well. If we do it like that we can version based on R2R module version (it's already what we do in `DebugInfo.cs`).

Reply:
I agree with adding `GCData` offset. Just wonder if 16-bit is enough. Is that just 256 locals each with 256 fields, or some variation of that?